src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -74,14 +75,18 @@
  * can be {@linkplain MemorySession#isAccessibleBy(Thread) accessed} by multiple threads. This might be useful when
  * multiple threads need to access the same memory segment concurrently (e.g. in the case of parallel processing).
  * Moreover, a shared arena can be closed by any thread.
+ * @implSpec
```

Comment:
This is a good point - but is only true for `allocate` I think? For `close` we need to make sure that owner thread is respected. So perhaps more specific implSpec on these two methods might do the job?